lib/skywalking/util.lua

```diff
@@ -1,34 +1,92 @@
--- 
+--
 -- Licensed to the Apache Software Foundation (ASF) under one or more
 -- contributor license agreements.  See the NOTICE file distributed with
 -- this work for additional information regarding copyright ownership.
 -- The ASF licenses this file to You under the Apache License, Version 2.0
 -- (the "License"); you may not use this file except in compliance with
 -- the License.  You may obtain a copy of the License at
--- 
+--
 --    http://www.apache.org/licenses/LICENSE-2.0
--- 
+--
 -- Unless required by applicable law or agreed to in writing, software
 -- distributed under the License is distributed on an "AS IS" BASIS,
 -- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 -- See the License for the specific language governing permissions and
 -- limitations under the License.
--- 
+--
+local _M = {}
+
+-- for pure Lua
+local split = function(str, delimiter)
+    local t = {}
+    for substr in string.gmatch(str, "[^".. delimiter.. "]*") do
+        if substr ~= nil and string.len(substr) > 0 then
+            table.insert(t,substr)
+        end
+    end
+    return t
+end
+
+local timestamp = function()
+    local _, b = math.modf(os.clock())
+    if b==0 then
+        b='000'
+    else
+        b=tostring(b):sub(3,5)
+    end
+
+    return os.time() * 1000 + b
+end
+
+-- for Nginx Lua
+local ok, ngx_re = pcall(require, "ngx.re")
+if ok then
+    split = ngx_re.split
+    timestamp = function()
+        return ngx.now() * 1000
+    end
+end
+
+_M.split = split
+_M.timestamp = timestamp
 
-local Util = {}
 local MAX_ID_PART2 = 1000000000
 local MAX_ID_PART3 = 100000
-local SEQ = 1
+local metadata_buffer = ngx.shared.tracing_buffer
```

Comment:
I think this is incompatible for pure lua.

Reply:
I will remove it.